Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -332,11 +323,24 @@ class DatabaseContainer: NSPersistentContainer {
             }
             if !deletedObjectIds.isEmpty, let contexts = self?.allContext {
                 log.debug("Merging \(deletedObjectIds.count) deletions to contexts", subsystems: .database)
+                // Merging changes triggers DB observers to react to deletions
                 NSManagedObjectContext.mergeChanges(
                     fromRemoteContextSave: [NSDeletedObjectsKey: deletedObjectIds],
                     into: contexts
                 )
             }
+            // Finally reset states of all the contexts after batch delete and deletion propagation.
+            if let writableContext = self?.writableContext, let allContext = self?.allContext {
+                writableContext.invalidateCurrentUserCache()
+                writableContext.reset()
```

Comment:
Did you also tried reseting these context when a delete is performed on `write` function? (Not sure if this will do anything tho)

Reply:
Did not test that. Slightly afraid to do it because we use these contexts for DB observers. Thinking about letting it be 😄

Reply:
Yeah probably better 👍

Reply:
Out of curiosity, I tested it and it works. Maybe if we still have problems with this in the future we can go back here and apply the following change before the `CatabaseContainer.write` :
```swift
if !self.backgroundReadOnlyContext.deletedObjects.isEmpty {
      self.backgroundReadOnlyContext.reset()
 }
```